client/src/Pages/SignUp.js

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+const SignUp = (props) => {
+
+	// Initial student in case that you want to update a new student
+    const {initialPost = {id: null, 
+		username: "", 
+		email: ""}} = props;
+
+
+// We're using that initial student as our initial state                       
+const [user, setUser] = useState(initialPost);
+
+//create functions that handle the event of the user typing into the form
+const handleUsernameChange = (event) => {
+const username = event.target.value;
+setUser((user) => ({ ...user, username }));
+
+}
+
+const handleEmailChange = (event) => {
+const email = event.target.value;
+setUser((user) => ({ ...user, email }));
+
+}
+
+//A function to handle the post request
+const postUser = (newUser) => {
+return fetch('/api/users', {
+method: 'POST',
+headers: {'Content-Type': 'application/json'}, 
+body: JSON.stringify(newUser)
+}).then((response) => {
+return response.json()
+}).then((data) => {
+console.log("From the post ", data);
+props.saveUser(data);
```

Comment:
I don't see this prop defined. Same observation for `saveStudent`.

Similar note from @ndp for `savePost`